nncf/quantization/algorithms/weight_compression/torch_backend.py

```diff
@@ -0,0 +1,250 @@
+
+        for wc_params in weight_compression_parameters:
+            compression_config = wc_params.compression_config
+            if compression_config.mode not in [
+                CompressWeightsMode.INT8_ASYM,
+                CompressWeightsMode.INT8_SYM,
+                CompressWeightsMode.INT8,
+            ]:
+                raise ValueError(f"{compression_config.mode.value} is not suported.")
+
+            weight_node = get_weight_node(wc_params.node_with_weight, wc_params.weight_port_id, graph)
+            weight_name = weight_node.layer_attributes.name
+            module_name, weight_attr_name = split_weight_name(weight_name)
+            module = get_moodule_by_name(module_name, model)
+            weight = getattr(module, weight_attr_name)
+            if weight is None or not isinstance(weight, torch.nn.Parameter):
+                raise RuntimeError(f"Could not find a torch.nn.Parameter in the model by name {weight_name}.")
+
+            # calculates compressed weights and decompression parameters
+            compressed_weight = compress_weight(Tensor(weight), wc_params.reduction_axis, compression_config)
+
+            # sets compressed tensor
+            compressed_parameter = torch.nn.Parameter(compressed_weight.tensor.data, requires_grad=False)
+            setattr(module, weight_attr_name, compressed_parameter)
+
+            consumer_nodes = graph.get_next_nodes(weight_node)
+            if len(consumer_nodes) > 1:
+                for c_node in consumer_nodes:
+                    c_module = model.nncf.get_module_by_scope(Scope.from_str(c_node.layer_name))
+                    for name, param in c_module.named_parameters(recurse=False, remove_duplicate=False):
+                        if id(param) == id(weight):
+                            setattr(c_module, name, compressed_parameter)
+
+            # creates weight decompressor
+            decompressor = WeightsDecompressor(compressed_weight.scale.data, compressed_weight.zero_point.data)
+
+            # registry weight decompression module in the model
+            decompressor_name = f"weights_decompressor_{weight_node.node_name.replace('.', '_')}"
+            model.nncf.add_compression_module(decompressor_name, decompressor, compression_model_type)
+
+            # inserts the weight decompressor into the model as the post hook on the model weight
+            hook = ExternalOpCallHook(EXTERNAL_OP_STORAGE_NAME, model.nncf.get_tracing_context(), decompressor_name)
+            node_to_op_address_mapping = model.nncf.get_node_to_op_address_mapping()
+            model.nncf.insert_at_point(
+                PTInsertionPoint(TargetType.OPERATOR_POST_HOOK, node_to_op_address_mapping[weight_node.node_name]),
+                [hook],
+            )
+
+        # rebuild compression graph with weight decompression module
+        model.nncf.rebuild_graph()
+
+        return model
+
+    @staticmethod
+    def dump_parameters(
```

Comment:
`dump_parameters` looks like only OV specific function, should is it be a part of backend?

Reply:
What exactly is your proposal?

Reply:
I was going to suggest as for quantize but it's not passible  😞 
So make this function not abstract and override it only for OV.